mdpow/analysis.py

```diff
@@ -0,0 +1,362 @@
+# mdpow: analysis.py
+# 2021 Alia Lescoulie
+
+"""A set of objects for analyzing MDPOW simulations
+
+"""
+from __future__ import absolute_import
+
+import os
+
+from gromacs.utilities import in_dir
+import MDAnalysis as mda
+from MDAnalysis.analysis.dihedrals import calc_dihedrals
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.log import ProgressBar
+from MDAnalysis.exceptions import FileFormatWarning, NoDataError, MissingDataWarning, SelectionError
+import numpy as np
+import pandas as pd
+
+import six
+
+import logging
+
+logger = logging.getLogger('mdpow.analysis')
+
+
+class NoDataWarning(Exception):
+    pass
+
+
+class Ensemble(object):
+    """Collection of related MDAnalysis Universe objects
+
+    Stores systems produced by running mdpow-fep organized
+    by solvent, interaction, and lambda.
+
+    Given a mdpow simulation directory will load the MD
+    simulation files with the directory structure as keys.
+
+    Typical work flow::
+
+        ens = Ensemble(dirname='molecule')
+
+    """
+
+    def __init__(self, dirname=None, solvents=['octanol', 'water'], **kwargs):
+        """
+
+        """
+        self.num_systems = 0
+        self.ensemble = {}
+        self.system_keys = []
+        self.solvents = solvents
+
+        if dirname is None:
+            return
+
+        if not os.path.exists(dirname):
+            logger.error("Directory %s does not exist" % dirname)
+        self.ensemble_dir = dirname
+        self.build_ensemble()
+
+    def __repr__(self):
+        return "<Ensemble Containing %s Universes>", self.num_systems
+
+    def __len__(self):
+        return self.num_systems
+
+    def __getitem__(self, index):
+        """Allows dictionary like indexing"""
+        try:
+            item = self.ensemble[index]
+            return item
+        except KeyError:
+            logger.error(KeyError("Invalid system key"))
+
+    def __iter__(self):
+        """Returns list of systems with object iterated over"""
+        sys_list = []
+        for key in self.system_keys:
+            sys_list.append(self[key])
+        return sys_list
+
+    @staticmethod
+    def _load_dir_unv():
+        """Loads system in directory into an MDAnalysis Universe
+
+        logs warning if more than one topology is in directory. If
+        more than one trajectory attempts to load both of them
+        in a universe if that fail will try to load each individually"""
+        cur_dir = os.listdir(os.curdir)
+        trj = []
+        top = []
+        for file in cur_dir:
+            if file.endswith('.xtc'):
+                trj.append(file)
+            if file.endswith('.gro'):
+                top.append(file)
```

Comment:
For multiple xtc files it first tries to just load a Universe with the trajectories as a list so that it can handle multiple xtc files in a directory.

I'll have to attempt to implement the path suggestion.

As for being able to specify a topology from the equilibration, that's a good idea, I'll try adding it as a kwarg, because at the moment if you wanted to do that the only option is using `add_system` several times.